include/libcyphal/transport/transport.hpp

```diff
@@ -61,20 +61,61 @@ class ITransport : public IRunnable
     /// @return `nullopt` on successful set (or when node ID is the same).
     ///         Otherwise an `ArgumentError` in case of the subsequent calls or ID out of range.
     ///
-    CETL_NODISCARD virtual cetl::optional<ArgumentError> setLocalNodeId(const NodeId node_id) noexcept = 0;
-
-    CETL_NODISCARD virtual Expected<UniquePtr<IMessageRxSession>, AnyError> makeMessageRxSession(
-        const MessageRxParams& params) = 0;
-    CETL_NODISCARD virtual Expected<UniquePtr<IMessageTxSession>, AnyError> makeMessageTxSession(
-        const MessageTxParams& params) = 0;
-    CETL_NODISCARD virtual Expected<UniquePtr<IRequestRxSession>, AnyError> makeRequestRxSession(
-        const RequestRxParams& params) = 0;
-    CETL_NODISCARD virtual Expected<UniquePtr<IRequestTxSession>, AnyError> makeRequestTxSession(
-        const RequestTxParams& params) = 0;
-    CETL_NODISCARD virtual Expected<UniquePtr<IResponseRxSession>, AnyError> makeResponseRxSession(
-        const ResponseRxParams& params) = 0;
-    CETL_NODISCARD virtual Expected<UniquePtr<IResponseTxSession>, AnyError> makeResponseTxSession(
-        const ResponseTxParams& params) = 0;
+    virtual cetl::optional<ArgumentError> setLocalNodeId(const NodeId node_id) noexcept = 0;
+
+    /// @brief Makes a message receive (RX) session.
+    ///
+    /// Lifetime of the RX session must never outlive this transport interface.
+    ///
+    /// @param params Various initial parameters for the message RX session (like subject id and extent size).
+    /// @return Either a unique interface pointer to the message RX session, or an error.
+    ///
```

Comment:
```suggestion
```

Do these comments serve any purpose at all? It is evident from the type of the method that it takes params and returns either an interface unique ptr or an error, why spell it out? For better SNR we should only write things that are worth writing.

@thirtytwobits this is an example of an unhelpful doc comment; I suggest we avoid these.

Reply:
Yes, I need them for two purposes:
1. Please see that, in addition to "obvious" stuff, there is also extra note about lifetime of result sessions regarding parent transport.
2. Doxygen - docs generation is just bad without explicit mentioning of public functions.
3. Last but not least - there is less confusion in rules what we document and what we don't - it's much easier to go with "always document public stuff" comparing to fuzzy "it is evident from ..."

Reply:
My objection related only to the two lines I commented on -- the ones that begin with `@param` and `@return`; the rest is useful and should be kept. I think this addresses points 1 and 2.

Reply:
oh, ok, got it - will remove these pairs of lines.